L1Trigger/TrackFindingTracklet/interface/Settings.h

```diff
@@ -968,6 +981,44 @@ namespace trklet {
 
     double stripLength_PS_{0.1467};
     double stripLength_2S_{5.0250};
+
+    //Variable bin edges for 1 to 12 bins.
+    std::vector<std::vector<double>> varrinvbins_{
+        {{rinvcut()},                                                                                         //1  Bin
+         {0, rinvcut()},                                                                                      //2  Bins
+         {-0.003828, 0.003828, rinvcut()},                                                                    //3  Bins
+         {-0.004459, 0, 0.004459, rinvcut()},                                                                 //4  Bins
+         {-0.004774, -0.003141, 0.003141, 0.004774, rinvcut()},                                               //5  Bins
+         {-0.004968, -0.003828, 0, 0.003828, 0.004968, rinvcut()},                                            //6  Bins
+         {-0.005105, -0.004207, -0.002718, 0.002718, 0.004207, 0.005105, rinvcut()},                          //7  Bins
+         {-0.005210, -0.004459, -0.003435, 0, 0.003435, 0.004459, 0.005210, rinvcut()},                       //8  Bins
+         {-0.005284, -0.004640, -0.003828, -0.002421, 0.002421, 0.003828, 0.004640, 0.005284, rinvcut()},     //9  Bins
+         {-0.005347, -0.004774, -0.004099, -0.003140, 0, 0.003140, 0.004099, 0.004774, 0.005347, rinvcut()},  //10 Bins
+         {-0.005392, -0.004880, -0.004299, -0.003550, -0.002193, 0.002193, 0.003550, 0.004299, 0.004880, 0.005392, rinvcut()},  //11 Bins
+         {-0.005433, -0.004968, -0.004459, -0.003828, -0.002911, 0, 0.002911, 0.003828, 0.004459, 0.004968, 0.005433, rinvcut()}}};  //12 Bins
+
+    std::vector<std::vector<double>> overlapbins() {
```

Comment:
Can we delete function overlapbins() and simply get Settings to return the non-overlapping rinv bin boundaries and the value of overlapsize_?
I believe overlapbins() is only called by PurgeDuplicates::findOverlapRinvBins(), which I suspect could also be dramatically simplified. Doesn't it just need to do something like:

double rInv = trk->rinv();
unsigned int chosenBin = 0;
vector<unsigned int> chosenBins;
for (unsigned int i = 0; i < varrinvbins.size(); i++) {
  if (rinv > varrinvbins[i] - overlapsize_ && 
      rinv < varrinvbins[i] + overlapsize_) {
      chosenBins.push_back(i);
  }  
where here I assume varrinvbbins[0] = -rinvcut();

Reply:
overlapbins() has been deleted, and findOverlapRinvBins() has been simplified the way suggested.